src/OrleansTestingHost/DefaultSiloBuilderFactory.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Orleans.Hosting;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+
+namespace Orleans.TestingHost
+{
+    public class DefaultSiloBuilderFactory : ISiloBuilderFactory
+    {
+        public ISiloBuilder CreateSiloBuilder(string siloName, ClusterConfiguration clusterConfiguration)
```

Comment:
please make this virtual, otherwise derived types have to explicitly implement the interface just to override it, which doesn't play nicely with several levels of inheritance (and it's actually discouraged to override in that way)

Reply:
Upon reflection, I think I'll make this sealed and force custom factory builders to be in test code.  The reason for this is that it will be clearer in the test code how the factory is being used.  This hides too much.